src/asciidoc/docs/develop/app-dev-process.adoc

```diff
@@ -0,0 +1,161 @@
+To define this data, most Koji projects include a `koji.json` file at the root of the project directory.
+
+. Define the instructions for building your code and deploying your app on the Koji platform.
++
+Learn more: <<projects#develop, develop key>>, <<projects#deploy, deploy key>>
+
+. If the app has customization options for creators, define the default values for those options and any placeholder values for new, customized versions.
++
+Learn more: <<projects#remixData, remixData key>>, <<projects#@@initialTransform, @@initialTransform key>>
+
+. Configure the entitlements for the Koji platform features in your app.
++
+Learn more: <<entitlements#>>
+
+=== Add @withkoji/core to the frontend
+
+Follow these steps to use the @withkoji/core package in the frontend of your app.
+
+[TIP]
+====
+Before you start developing your code, check for outdated dependencies in the forked app.
+
+* If the forked app uses deprecated Koji packages, migrate your project to use the @withkoji/core package.
++
+ Learn more: <<migrate-koji-core#, Do I need to migrate?>>
+
+* Check for other outdated dependencies in the forked app, and update them if necessary.
++
+ Learn more: <<work-locally#_installing_the_dependencies, Installing dependencies>>
+====
+
+. In your frontend directory, install or update the @withkoji/core package.
++
+[source, Bash]
+npm install @withkoji/core
+
+. In your frontend code, import the package.
++
+[source,Javascript]
+import Koji from '@withkoji/core';
+
+. Use the `Koji.config` function to initialize the package with your app's configuration data.
++
+Ideally, your code will initialize the package one time, before any data in the application is rendered.
+For example:
++
+[source,Javascript]
+----
+// Initialize
+Koji.config(require('././koji.json'));
+
+// render
+render();
+----
+
+. Use the `Koji.ready` function to indicate that the application is ready to start receiving events.
+
+. Implement the desired core functionality.
+For example, you can provide customization options so creators can add their own content and branding to the app before adding it to their profile.
++
+Learn more: <<core-frontend-overview#>>, <<capture-user-input#>>, <<thumbnail-image#>>
+
+=== Add @withkoji/core to the backend
+
+If your app has a backend service, follow these steps to use the @withkoji/core package in it.
+
+. In your backend directory, install or update the @withkoji/core package.
++
+[source, Bash]
+npm install @withkoji/core
+
+. Make sure the `koji.json` file will be bundled with the backend service during deployment.
++
+TIP: In most frameworks, this file won’t be bundled by default because it is not in the same folder as the backend code.
+If the file is not bundled, the backend service will not be initialized correctly and the deployment will fail.
+To solve this problem, you can use a tool such as https://www.npmjs.com/package/babel-plugin-inline-json-import[babel-plugin-inline-json-import].
```

Comment:
Not for adding to this topic, but these are questions that come to mind:
* Why do they need an external tool to do this?
* How do they use this tool to do this?
* Why do we recommend this specific tool?
* What are other alternatives?

I tend to veer away from recommending third-party apps/solutions unless they can be vetted.

Reply:
As I understand it, we are recommending a file structure that has a potential gotcha for most frameworks, and all apps will need to use something to build their code. This problem is probably less important to call out when someone is using an internally developed app or scaffold as their starting point, because those have already implemented this solution. I'll try to clarify this step a bit, and maybe Sean will have additional feedback to answer some of these questions.